### PR TITLE
ci: deprecate old dev releases on publish

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds a step to the publish dev CI that automatically deprecates all dev versions before publishing a new one so that you don't have to manually do that yourselves every 12 hours. Here are the npm docs on [npm-deprecate](https://docs.npmjs.com/cli/v7/commands/npm-deprecate) and you can test the version range [here](https://semver.npmjs.com/)

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: .github/workflows/publish-dev.yml [hunk 1]
```diff
@@ -21,6 +21,11 @@ jobs:
         # Remove --no-optional after Node 16 upgrade
         run: npm ci --ignore-scripts --no-optional
 
+      - name: Deprecate old versions
+        run: npm deprecate discord.js@"~13.0.0-dev" "No longer supported" || true
```
Comment (vladfrangu) [inline .github/workflows/publish-dev.yml:25]: Can't something like this be done, so it auto deprecates any valid dev version from the package.json?

```suggestion
        run: npm deprecate discord.js@"~$(jq --raw-output '.version' package.json)" "No longer supported" || true
```

**Review by vladfrangu — DISMISSED** [review]

File: .github/workflows/publish-dev.yml (hunk 1 above)
Comment (ImRodry) [reply .github/workflows/publish-dev.yml:25]: I thought this wasn’t gonna be used after v13 but if it will then yeah I think that can work

Comment (vladfrangu) [reply .github/workflows/publish-dev.yml:25]: Oh, gotcha

Comment (iCrawl) [inline .github/workflows/publish-dev.yml:25]:
```suggestion
        run: npm deprecate discord.js@"~13.0.0-dev" "no longer supported" || true
```

Comment (iCrawl) [reply .github/workflows/publish-dev.yml]: For now, yes.

**Review by iCrawl — APPROVED** [review]

**Pull request merged**